pkg/controller/disruption/disruption.go

```diff
@@ -174,14 +182,15 @@ func NewDisruptionController(
 // resources directly and only fall back to the scale subresource when needed.
 func (dc *DisruptionController) finders() []podControllerFinder {
 	return []podControllerFinder{dc.getPodReplicationController, dc.getPodDeployment, dc.getPodReplicaSet,
-		dc.getPodStatefulSet, dc.getScaleController}
+		dc.getPodStatefulSet, dc.getPodDaemonSet, dc.getScaleController}
```

Comment:
I admit I'm surprised we have added this method in general.  Where is the fallback logic to use scale for this case?  getExpectedScale() doesn't implement a generic fallback, so that means this logic is not extensible and that's not great.  If there is a generic fallback, that would imply this is an optimization only (in which case, I would have expected the bug report to be "controller lookups for daemonsets are slow and this improves performance", which doesn't match what you are reporting).

Reply:
Is the generic lookup dc.getScaleController?  Is Scale not implemented for DaemonSets?

Reply:
>Is Scale not implemented for DaemonSets?

It is not.

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/apps/v1/types.go#L636

Reply:
Oh.

Oh....

Reply:
Hrm.  That's really ugly.  It should support read scale but not write scale, or reject write scale.